Sources/Swift/SentryCrash/SentryCrashWrapper.swift

```diff
@@ -0,0 +1,291 @@
+        // isiOSAppOnMac and isMacCatalystApp are mutually exclusive, so we only set one of them.
+        if #available(iOS 14.0, macOS 11.0, watchOS 7.0, tvOS 14.0, *) {
+            if self.processInfoWrapper.isiOSAppOnMac {
+                runtimeContext["name"] = "iOS App on Mac"
+                runtimeContext["raw_description"] = "ios-app-on-mac"
+            }
+        }
+        
+        if #available(iOS 13.0, macOS 10.15, watchOS 6.0, tvOS 13.0, *) {
+            if self.processInfoWrapper.isMacCatalystApp {
+                runtimeContext["name"] = "Mac Catalyst App"
+                runtimeContext["raw_description"] = "raw_description"
+            }
+        }
+        
+        if !runtimeContext.isEmpty {
+            scope.setContext(value: runtimeContext, key: "runtime")
+        }
+    }
+    
+    private func getOSName() -> String {
+#if os(macOS) || targetEnvironment(macCatalyst)
+        return "macOS"
+#elseif os(iOS)
+        return "iOS"
+#elseif os(tvOS)
+        return "tvOS"
+#elseif os(watchOS)
+        return "watchOS"
+#elseif (swift(>=5.9) && os(visionOS))
+        return "visionOS"
+#else
+        return "Unknown"
+#endif
+    }
+    
+    private func getOSVersion() -> String {
+        // For MacCatalyst the UIDevice returns the current version of MacCatalyst and not the
+        // macOSVersion. Therefore we have to use NSProcessInfo.
+#if (os(iOS) || os(tvOS) || (swift(>=5.9) && os(visionOS))) && !SENTRY_NO_UIKIT && !targetEnvironment(macCatalyst)
+        return Dependencies.uiDeviceWrapper.getSystemVersion()
+#else
+        let version = ProcessInfo.processInfo.operatingSystemVersion
+        return "\(version.majorVersion).\(version.minorVersion).\(version.patchVersion)"
+#endif // (os(iOS) || os(tvOS) || (swift(>=5.9) && os(visionOS))) && !SENTRY_NO_UIKIT && !targetEnvironment(macCatalyst)
+    }
+    
+    private func isSimulator() -> Bool {
+#if targetEnvironment(simulator)
+        return true
+#else
+        return false
+#endif // targetEnvironment(simulator)
+    }
+    
+    private func getDeviceFamily(from systemName: String) -> String? {
+        let family = systemName.components(separatedBy: .whitespacesAndNewlines).first
+#if targetEnvironment(macCatalyst)
+        // This would be iOS. Set it to macOS instead.
+        return "macOS"
+#else
+        return family
+#endif // targetEnvironment(macCatalyst)
+    }
+    
+    private func setScreenDimensions(_ deviceData: inout [String: Any]) {
+        // The UIWindowScene is unavailable on visionOS
+#if (os(iOS) || os(tvOS)) && !SENTRY_NO_UIKIT
+        if let appWindows = SentryDependencyContainerSwiftHelper.windows(),
+           let appScreen = appWindows.first?.screen {
+            deviceData["screen_height_pixels"] = appScreen.bounds.size.height
+            deviceData["screen_width_pixels"] = appScreen.bounds.size.width
+        }
+#endif // (os(iOS) || os(tvOS)) && !SENTRY_NO_UIKIT
+    }
+    
+    // MARK: - Private Methods
```

Comment:
`l`: That needs to move up before ` private func enrichScopeWithOSData(_ scope: Scope, systemInfo: [String: Any])`.